src/cascadia/TerminalApp/TabManagement.cpp

```diff
@@ -905,10 +908,29 @@ namespace winrt::TerminalApp::implementation
             {
                 _TitleChangedHandlers(*this, tab.Title());
             }
+
+            auto tab_impl = _GetTerminalTabImpl(tab);
+            if (tab_impl)
+            {
+                auto profile = tab_impl->GetFocusedProfile();
+                _UpdateBackground(profile);
+            }
         }
         CATCH_LOG();
     }
 
+    void TerminalPage::_UpdateBackground(const winrt::Microsoft::Terminal::Settings::Model::Profile& profile)
+    {
+        if (profile)
+        {
+            auto controlSettings = TerminalSettings::CreateWithProfile(_settings, profile, *_bindings);

```

Comment:
may be able to skip building the TerminalSettings here and just go straight for `profile.DefaultAppearance()`, and get `BackgroundImagePath()` out of that